modules/deploy/pages/deployment-option/self-hosted/manual/production/production-readiness.adoc

```diff
@@ -0,0 +1,534 @@
+```
+
+Ensure that xref:reference:rpk/rpk-iotune.adoc[`rpk iotune run`] has been run on all hosts. Ensure that the mountpoint listed in this configuration file matches the mountpoint for Redpanda's data directory, usually configured at `/var/lib/redpanda`. See xref:deploy/deployment-option/self-hosted/manual/production/production-deployment.adoc#generate-optimal-io-configuration-settings[Generate optimal I/O configuration settings].
+
+Example:
+```bash
+cat /etc/redpanda/io-config.yaml
+disks:
+  - mountpoint: /mnt/vectorized
+    read_iops: 413115
+    read_bandwidth: 1882494592
+    write_iops: 182408
+    write_bandwidth: 788050688
+```
+
+[[disk-perf]]
+=== Run self test to check disk performance
+
+Run the self test to ensure that disk performance is within an acceptable range.
+
+Example:
+```bash
+rpk cluster self-test status
+```
+
+Output:
+```bash
+NODE ID: 1 | STATUS: IDLE
+=========================
+NAME        512KB sequential r/w throughput disk test
+INFO        write run
+TYPE        disk
+TEST ID     e13b2c93-2417-458b-87be-fac409089513
+TIMEOUTS    0
+DURATION    30000ms
+IOPS        984 req/sec
+THROUGHPUT  492.1MiB/sec
+LATENCY     P50     P90     P99     P999    MAX
+            4095us  4095us  4351us  4607us  5119us
+```
+
+[[hostnames-interfaces]]
+=== Advertised hostnames use correct interfaces
+
+Ensure that the advertised hostnames are operating on the correct network interfaces.
+
+Example:
+```bash
+grep -A1 advertised /etc/redpanda/redpanda.yaml
+    advertised_kafka_api:
+    -   address: 54.244.110.128
+--
+    advertised_rpc_api:
+        address: 172.31.40.237
+```
+
+[[continuous-db]]
+=== Continuous data balancing is configured
+
+Ensure that xref:manage:cluster-maintenance/continuous-data-balancing.adoc[Continuous Data Balancing] is configured and enabled.
+
+Example: 
+```bash
+rpk cluster config get partition_autobalancing_mode
+```
+
+Output:
+```bash
+continuous
+```
+
+[[debug-bundle]]
+=== Generate a debug bundle
+
+Ensure that you’re able to generate a debug bundle from each host and upload to Redpanda Support.
+
+```bash
+sudo rpk debug bundle
+Creating bundle file...
+Debug bundle saved to '1711372017-bundle.zip'
+```
+
+[[topic-rf]]
+=== Topic replication factor
+
+Ensure that the topics replication factor are greater than one.
+
+Example:
+```bash
+rpk topic list
+```
+
+Output:
+```bash
+NAME   PARTITIONS  REPLICAS
+bad    1           1
+good   1           3
+```
+
+[[maintenance-mode]]
+=== Brokers in maintenance mode
+
+Ensure that no brokers are in maintenance mode.
+
+Example:
+
+```bash
+rpk cluster maintenance status
+```
+
+Output:
+
+```bash
+NODE-ID  ENABLED  FINISHED  ERRORS  PARTITIONS  ELIGIBLE  TRANSFERRING  FAILED
+1        false    -         -       -           -         -             -
+2        false    -         -       -           -         -             -
+3        false    -         -       -           -         -             -
+```
+
```

Comment:
See also:
- https://docs.redpanda.com/current/manage/node-management/#place-a-broker-in-maintenance-mode